src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -257,6 +266,19 @@ public Instance(
         } else {
             log.info("collect_default_jvm_metrics is false - not collecting default JVM metrics");
         }
+        //create bean to track data of this instance
+        JMXbean = beanManager.createJMXBean(this);
+    }
+
+    public void updateBeanInfo()throws Exception {
+        if (JMXbean != null){
+            refreshBeansList();
+            getMatchingAttributes();
+            JMXbean.setBeanCount(beans.size());
+            JMXbean.setAttributeCount(matchingAttributes.size());
+            JMXbean.setMetricCount(getMetrics().size());
```

Comment:
Similar to my comment above, `getMetrics` is a very expensive call, we should not be adding more calls to it.